File: 프로그래머스/unrated/138477. 명예의 전당 （1）/명예의 전당 （1）.py
def solution(k, score):
    answer = []
    k_score=[]
    
    for i in score:
        k_score.append(i)
        k_score.sort(reverse=True)
        
        if len(k_score) > k:
            k_score.pop()
        
        answer.append(k_score[-1])
    return answer

File: 프로그래머스/1/138477. 명예의 전당 （1）/명예의 전당 （1）.java
import java.util.*;

class Solution {
    public int[] solution(int k, int[] score) {
        int[] answer = new int[score.length];
        
        ArrayList<Integer> list = new ArrayList<>();
        
        for (int i = 0;i < score.length;i++) {
            list.add(score[i]);
            if (list.size() > k) {
                int min = Collections.min(list);
                int indexMin = list.indexOf(min);
                
                list.remove(indexMin);
            }
            answer[i] = Collections.min(list);
        }
        
        return answer;
    }
}